test/run-unit-tests.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+import dataclasses
+import glob
+import os
+import subprocess
+import sys
+try:
+    import colorama as c
+    GREEN = c.Fore.GREEN
+    YELLOW = c.Fore.YELLOW
+    RED = c.Fore.RED
+    RESET_ALL = c.Style.RESET_ALL
+    BRIGHT = c.Style.BRIGHT
+except ImportError:
+    GREEN = YELLOW = RED = RESET_ALL = BRIGHT = ''
+
+@dataclasses.dataclass
+class Total:
+    total:int
+    good:int = 0
+    skip:int = 0
+    fail:int = 0
+
+tests = glob.glob('/usr/lib/systemd/tests/test-*')
+total = Total(total=len(tests))
+for test in tests:
+    name = os.path.basename(test)
+
+    ex = subprocess.run(test, stdin=subprocess.DEVNULL, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if ex.returncode == 0:
+        print(f'{GREEN}PASS: {name}{RESET_ALL}')
+        total.good += 1
```

Comment:
Green/RED no no

Reply:
unittest it much better than this

Reply:
>- @ssahani's test runner [2]
  It uses the unittest library and the test suite was implented as a class, and
  doesn't implement any of the functions listed above.

every test function is generated dynamically if that is a test and  unit test is standard.